lib/doorkeeper/exception_raiser.rb

```diff
@@ -0,0 +1,32 @@
+module Doorkeeper
+  class ExceptionRaiser
+    attr_reader :error
+
+    def initialize(error)
+      @error = error
+    end
+
+    def raise_if_handled
+      return unless raise_errors?
+
+      raise_invalid_token if error.is_a? OAuth::InvalidTokenResponse
```

Comment:
a minor thing, but `raise_invalid_token_error` might be a more descriptive name for this method